common/src/main/java/com/microsoft/identity/common/internal/broker/PackageHelper.java

```diff
@@ -160,4 +159,53 @@ public static String getBrokerRedirectUrl(final String packageName, final String
         }
         return "";
     }
+
+    /**
+     * Helper method to get signatures in a back-compatible way
+     *
+     * @param packageInfo A packageInfo instance with the flag PackageManager.GET_SIGNING_CERTIFICATES/PackageManager.GET_SIGNATURES set
+     * @return Signature[] or null
+     */
+    public static Signature[] getSignatures(final PackageInfo packageInfo) {
+        if (packageInfo == null) {
+            return null;
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            if (packageInfo.signingInfo == null) {
+                return null;
+            }
+            if (packageInfo.signingInfo.hasMultipleSigners()) {
+                return packageInfo.signingInfo.getApkContentsSigners();
+            } else {
+                return packageInfo.signingInfo.getSigningCertificateHistory();
+            }
+        }
+
+        return packageInfo.signatures;
+    }
+
+    public static int getPackageManagerFlag() {
```

Comment:
Method name is too vague - maybe `getPackageManagerSignaturesFlag()`

Method shouldn't be `public` - I would expect maybe the following?
```java
@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Reply:
Callers/clients do not necessarily have to be tests so I am of the opinion of maintaining it as `public`